src/shadow-embed.js

```diff
@@ -111,6 +115,39 @@ function createShadowRootPolyfill(hostElement) {
 
 
 /**
+ * Determines if value is actually a `ShadowRoot` node.
+ * @param {*} value
+ * @return {boolean}
+ */
+export function isShadowRoot(value) {
+  if (!value) {
+    return false;
+  }
+  // Node.nodeType == DOCUMENT_FRAGMENT to speed up the tests. Unfortunately,
+  // nodeType of DOCUMENT_FRAGMENT is used currently for ShadowRoot nodes.
+  if (value.tagName == 'I-AMP-SHADOW-ROOT') {
+    return true;
+  }
+  return (value.nodeType == /* DOCUMENT_FRAGMENT */ 11 &&
+      Object.prototype.toString.call(value) === '[object ShadowRoot]');
+}
+
+
+/**
+ * Return shadow root for the specified node.
+ * @param {!Node} node
+ * @return {?ShadowRoot}
+ */
+export function getShadowRootNode(node) {
+  if (isShadowDomSupported() && Node.prototype.getRootNode) {
```

Comment:
`isShadowDomSupported()` seems somewhat redundant, I assume if browser supports getRootNode, they satisfy the condition for `isShadowDomSupported()` anyway.